frontend/src/views/Nodes/components/TNodesOverviewContent.vue

```diff
@@ -0,0 +1,399 @@
+                        query: {
+                          cluster: $route.query.cluster,
+                          namespace: $route.query.namespace,
+                          uid: $route.query.uid,
+                        },
+                        params: {
+                          node: $route.params.node,
+                          service: service.metadata.id,
+                        },
+                      })
+                  "
+                  >View Log</t-button
+                >
+              </li>
+            </ul>
+          </t-group-animation>
+        </div>
+      </template>
+    </t-watch>
+  </div>
+</template>
+
+<script lang="ts">
+import TBreadcrumbs from "@/components/TBreadcrumbs.vue";
+import { computed, toRef } from "@vue/reactivity";
+import { useRoute, useRouter } from "vue-router";
+import TButton from "@/components/common/Button/TButton.vue";
+import TNodesContentType from "./TNodesContentType.vue";
+import TTag from "@/components/common/Tag/TTag.vue";
+import TStatus from "@/components/common/Status/TStatus.vue";
+import { getContext } from "@/context";
+import TWatch from "@/components/common/Watch/TWatch.vue";
+import TGroupAnimation from "@/components/common/Animation/TGroupAnimation.vue";
+export default {
+  components: {
+    TBreadcrumbs,
+    TButton,
+    TNodesContentType,
+    TTag,
+    TStatus,
+    TWatch,
+    TGroupAnimation,
+  },
+  props: {
+    items: {
+      type: Object,
+      required: true,
+    },
+  },
+  setup(props) {
+    const nodesList = toRef(props, "items");
+    const route = useRoute();
+    const router = useRouter();
+    const rebootNode = async () => {
+      router.push({
+        query: {
+          modal: "reboot",
+          node: route.params.node,
+          ...route.query,
+        },
+      });
+    };
+
+    const resetNode = async () => {
+      router.push({
+        query: {
+          modal: "reset",
+          node: route.params.node,
+          ...route.query,
+        },
+      });
+    };
+    const getItem = () => {
+      return nodesList.value.filter((item) => {
+        let addr;
+        for (const a of item.status.addresses) {
+          if (a.type == "InternalIP") {
+            addr = a.address;
+          }
+
+          if (a.type == "ExternalIP") {
+            addr = a.address;
+            break;
+          }
+        }
+        return addr == route.params.node;
+      });
+    };
+
+    return {
+      getContext,
+      rebootNode,
+      resetNode,
+      item: computed(() => getItem()),
+      roles: computed(() => {
+        const roles: any = [];
+        for (const label in getItem()[0]?.metadata?.labels) {
+          if (label.indexOf("node-role.kubernetes.io/") != -1) {
+            roles.push(label.split("/")[1]);
+          }
+        }
+        return roles;
+      }),
+      status: computed(() => {
+        const conditions: any = getItem()[0]?.status?.conditions;
+        if (!conditions) return "Not Ready";
```

Comment:
This method can be extracted and reused in several components and can utilize `constants/index.ts`.